tests/users/test_api_resources.py

```diff
@@ -343,6 +344,39 @@ def test_user_registration_with_taken_email(self, send_email_function):
             )
             self.assertEqual(response.status_code, 409)
 
+    @patch("flask_mail._MailMixin.send", side_effect=mail_send_mocked)
+    def test_user_registration_race(self, send_email_function):
+        user = UserModel.query.first()
+        email = user.email
+        results = []
+
+        def request(x):
+            response = self.client.post(
+                "/register",
+                data=json.dumps(
+                    dict(
+                        name=user1["name"],
+                        username=user1["username"],
+                        password=user1["password"],
+                        email=user1["email"],
+                        terms_and_conditions_checked=user1[
+                            "terms_and_conditions_checked"
+                        ],
+                    )
+                ),
+                follow_redirects=True,
+                content_type="application/json",
+            )
+            results.append(json.loads(response.get_data(as_text=True)).get("message"))
+
+        with ThreadPoolExecutor(max_workers=10) as executor:
+            executor.map(request, [0, 0])
+            executor.shutdown(wait=True)
+        self.assertTrue(
+            USER_USES_AN_EMAIL_ID_THAT_ALREADY_EXISTS.get("message") in results
```

Comment:
Here, why `USER_USES_A_USERNAME_THAT_ALREADY_EXISTS` is not raised?

Reply:
If the email and username both are same only the email exception is raised.